turbinia/evidence.py

```diff
@@ -303,21 +313,24 @@ class GoogleCloudDiskRawEmbedded(GoogleCloudDisk):
   """
 
   def __init__(self, embedded_path=None, *args, **kwargs):
-    """Initialization for Google Cloud Disk."""
+    """Initialization for Google Cloud Disk containing a raw disk image."""
     self.embedded_path = embedded_path
     super(GoogleCloudDiskRawEmbedded, self).__init__(*args, **kwargs)
 
+    # This Evidence needs to have a GoogleCloudDisk as a parent
+    self.context_dependent = True
+
   def _preprocess(self):
-    self.local_path = google_cloud.PreprocessAttachDisk(self.disk_name)
-    self.loopdevice_path = mount_local.PreprocessLosetup(self.local_path)
+    # TODO, after we make all Disks Evidence try to mount their filesystem,
+    # the following mounting will be done in the parent evidence.
+    # and we'll set mount_path as the path to the mounted embedded disk image.
     self.mount_path = mount_local.PreprocessMountDisk(
-        self.loopdevice_path, self.mount_partition)
-    self.local_path = os.path.join(self.mount_path, self.embedded_path)
+        self.parent_evidence.device_path, 1)
+    self.device_path = os.path.join(self.mount_path, self.embedded_path)
+    self.local_path = self.device_path
 
   def _postprocess(self):
-    google_cloud.PostprocessDetachDisk(self.disk_name, self.local_path)
     mount_local.PostprocessUnmountPath(self.mount_path)
```

Comment:
We use the parent evidence device_path to know how to mount it and expose the embed image. Resulting path is saved in self.mount_path
On post processing, we only need to unmount self.mount_path here

Reply:
Right, but there are two different mounts that can happen, one for the outer or parent disk, and a separate one for the embedded image.  In order to scan the embedded image, we're going to have to mount it somewhere as well, so I think the parent evidence should be mounted here at `evidence.parent_evidence.mount_path`, and the embedded evidence should be mounted at `evidence.mount_path`.  This will have to happen once we isolate the processors to their respective evidence types.

Reply:
Oh I see,  I forgot we still don't have all "disk-like" evidence do mounting/unmounting all the time.
Implementing it here